File: sdk/python/spiderweb_protocol/examples/control_handshake.py
import asyncio

from spiderweb_protocol import ControlClient, TextTransport


class DemoTransport(TextTransport):
    async def send_text(self, text: str) -> None:
        print("send:", text)

    async def receive_text(self) -> str:
        raise RuntimeError("provide a real websocket transport adapter")

    async def close(self) -> None:
        return None


async def main() -> None:
    client = ControlClient(DemoTransport())
    await client.negotiate_version("example-version")


asyncio.run(main())

File: sdk/typescript/spiderweb-protocol/examples/control-handshake.ts
import {
  ControlClient,
  WebSocketTextTransport,
} from "../src/index.js";

async function main(): Promise<void> {
  const transport = await WebSocketTextTransport.connect("ws://127.0.0.1:18790/");
  const client = new ControlClient(transport);

  console.log(await client.negotiateVersion("example-version"));
  console.log(await client.connect("example-connect"));

  await transport.close();
}

void main();
